### ISSUE TITLE
Remove overlay.OverlayAddr

### ISSUE THREAD

**Issue by scrye** [scionproto/scion#3297]
We currently support only two overlays, UDP/IPv4 and UDP/IPv6, which are both covered by the Go standard library type `net.UDPAddr`. However, to model these addresses we instead use type `overlay.OverlayAddr`. Migrating to `net.UDPAddr` yields several advantages.

Address construction becomes much simpler with `net.UDPAddr`. Currently, constructing an address calls a few functions and introduces a branch:
```
a, err := overlay.NewOverlayAddr(
    addr.HostFromIP(net.IP{10, 0, 0, 1}),
    addr.NewL4UDPInfo(80),
}
if err != nil {
    // error branch
}
```
versus
```
a := &net.UDPAddr{
    IP: net.IP{10, 0, 0, 1},
    Port: 80,
}
```

Access to addresses also becomes simpler and easier to reason about. E.g., for
```
var a *overlay.OverlayAddr
ip := a.L3().IP()
checkIfValidIP(ip)
port := a.L4().Port()
```
we need to check what `L3` returns, whether it can be nil, and whether it even is convertible to an IP (as there are types implementing the interface that do not return well-formed IP addresses, e.g., `addr.HostSVC`). While they are currently disallowed by the `OverlayAddr` constructor, this is not safe against future refactors, so code must always guard against implementations that do not return IPs. Developers must also check both method `L3` and the constructor to have a clear idea of what is correct use.

Using the standard library type directly is simpler:
```
var a *net.UDPAddr
DoSomething(a.IP)
DoSomethingElse(a.Port)
```
and supports all current uses of `OverlayAddr` in the code base.

Because `overlay.OverlayAddr` contains `interface` fields, serializing and deserializing the type is complex, which led to the creation of a multitude of other address types (that we should also remove in the future).

Additionally, due to how methods `Port()` and `IP()` work, it is a non-trivial task to add a new overlay even if we keep `overlay.OverlayAddr`.  Using `net.Addr` instead would mean that code that doesn't care about address internals can/should never look into it, and only overlay conn operations (creation, reads, writes) need to care about the address internals. This pattern works very well in the standard library where each `Conn` implementation attempts to type assert to its corresponding address type. This can also work for us, without exposing `L3` and `L4` (because as soon as an API exposes something, some package somewhere will try to use it, thus making future refactors more difficult).

To summarize, moving from `overlay.OverlayAddr` to `net.UDPAddr` would eliminate several bug-prone caveats, and improve code readability and maintainability.
